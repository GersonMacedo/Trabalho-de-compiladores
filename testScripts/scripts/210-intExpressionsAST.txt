! -s TREE
! should print correctly the AST with the int expressions
program p ;
    var i1 : integer;
    var i2 : integer;
    var i3 : integer;
    var i4 : integer;
    var i5 : integer;
begin
    i1 := i1 + i2;
    i1 := i1 + i2 + i3;
    i1 := i1 - i2;
    i1 := i1 - i2 - i3;
    i2 := i1 + i2 - i3 + i4 - i5;

    i1 := i1 * i2;
    i1 := i1 * i2 * i3;
    i1 := i1 / i2;
    i1 := i1 / i2 / i3;
    i2 := i1 * i2 / i3 * i4 / i5;

    i1 := i1 * i2 + i3;
    i1 := i1 * (i2 + i3);
    i1 := i1 * i2 - i3;
    i1 := i1 * (i2 - i3);
    i1 := i1 / i2 + i3;
    i1 := i1 / (i2 + i3);
    i1 := i1 / i2 - i3;
    i1 := i1 / (i2 - i3);

    i2 := i1 + i2 * i3;
    i2 := (i1 + i2) * i3;
    i2 := i1 - i2 * i3;
    i2 := (i1 - i2) * i3;
    i2 := i1 + i2 / i3;
    i2 := (i1 + i2) / i3;
    i2 := i1 - i2 / i3;
    i2 := (i1 - i2) / i3;
end .