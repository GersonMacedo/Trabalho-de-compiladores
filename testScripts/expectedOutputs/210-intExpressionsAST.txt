
Performing up to syntactic analysis and printing the AST tree at testScripts/scripts/210-intExpressionsAST.txt
No errors found

Programa 'p'
Declarações:
    'i1': INTEGER
    'i2': INTEGER
    'i3': INTEGER
    'i4': INTEGER
    'i5': INTEGER
Corpo:
    'i1' :=
        ADD
            'i1'
            'i2'
    'i1' :=
        ADD
            ADD
                'i1'
                'i2'
            'i3'
    'i1' :=
        SUB
            'i1'
            'i2'
    'i1' :=
        SUB
            SUB
                'i1'
                'i2'
            'i3'
    'i2' :=
        SUB
            ADD
                SUB
                    ADD
                        'i1'
                        'i2'
                    'i3'
                'i4'
            'i5'
    'i1' :=
        MULT
            'i1'
            'i2'
    'i1' :=
        MULT
            MULT
                'i1'
                'i2'
            'i3'
    'i1' :=
        DIV
            'i1'
            'i2'
    'i1' :=
        DIV
            DIV
                'i1'
                'i2'
            'i3'
    'i2' :=
        DIV
            MULT
                DIV
                    MULT
                        'i1'
                        'i2'
                    'i3'
                'i4'
            'i5'
    'i1' :=
        ADD
            MULT
                'i1'
                'i2'
            'i3'
    'i1' :=
        MULT
            'i1'
            ADD
                'i2'
                'i3'
    'i1' :=
        SUB
            MULT
                'i1'
                'i2'
            'i3'
    'i1' :=
        MULT
            'i1'
            SUB
                'i2'
                'i3'
    'i1' :=
        ADD
            DIV
                'i1'
                'i2'
            'i3'
    'i1' :=
        DIV
            'i1'
            ADD
                'i2'
                'i3'
    'i1' :=
        SUB
            DIV
                'i1'
                'i2'
            'i3'
    'i1' :=
        DIV
            'i1'
            SUB
                'i2'
                'i3'
    'i2' :=
        ADD
            'i1'
            MULT
                'i2'
                'i3'
    'i2' :=
        MULT
            ADD
                'i1'
                'i2'
            'i3'
    'i2' :=
        SUB
            'i1'
            MULT
                'i2'
                'i3'
    'i2' :=
        MULT
            SUB
                'i1'
                'i2'
            'i3'
    'i2' :=
        ADD
            'i1'
            DIV
                'i2'
                'i3'
    'i2' :=
        DIV
            ADD
                'i1'
                'i2'
            'i3'
    'i2' :=
        SUB
            'i1'
            DIV
                'i2'
                'i3'
    'i2' :=
        DIV
            SUB
                'i1'
                'i2'
            'i3'
